query/venues.js

```diff
@@ -0,0 +1,117 @@
+const peliasQuery = require('pelias-query');
+const defaults = require('./search_defaults');
+const logger = require('pelias-logger').get('api');
+const _ = require('lodash');
+const check = require('check-types');
+
+//------------------------------
+// general-purpose search query
+//------------------------------
+const venuesQuery = new peliasQuery.layout.VenuesQuery();
+
+// scoring boost
+venuesQuery.score( peliasQuery.view.focus_only_function( peliasQuery.view.phrase ) );
+// --------------------------------
+
+// non-scoring hard filters
+venuesQuery.filter( peliasQuery.view.boundary_country );
+venuesQuery.filter( peliasQuery.view.boundary_circle );
+venuesQuery.filter( peliasQuery.view.boundary_rect );
+venuesQuery.filter( peliasQuery.view.sources );
+// --------------------------------
+
+const adminLayers = ['neighbourhood', 'borough', 'city', 'county', 'state', 'country'];
+
+/**
+  map request variables to query variables for all inputs
+  provided by this HTTP request.  This function operates on res.data which is the
+  Document-ified placeholder repsonse.
+**/
+function generateQuery( clean ){
+  const vs = new peliasQuery.Vars( defaults );
+
+  const logParts = ['query:venues', 'parser:libpostal'];
+
+  // sources
+  if( !_.isEmpty(clean.sources) ) {
+    vs.var( 'sources', clean.sources);
+    logParts.push('param:sources');
+  }
+
+  // size
+  if( clean.querySize ) {
+    vs.var( 'size', clean.querySize );
+    logParts.push('param:querySize');
+  }
+
+  const mostGranularLayer = adminLayers.find(layer => {
+    return _.has(clean.parsed_text, layer);
+  });
+
+  if (mostGranularLayer) {
+    vs.var('input:query', clean.parsed_text[mostGranularLayer]);
+  }
+
+
+  // focus point
+  if( check.number(clean['focus.point.lat']) &&
```

Comment:
Probably outside the scope of this PR, however, each time we make a copy of existing code I have an internal rule: _ if this is the 3rd copy, it's time to move it to a utility function_.
We have the `focus.point` and `boundary.rect` handling in each of our query files: search, autocomplete, reverse, structured, search_original, and now this one. Would be great if we handled them similar to the existing `query\text_parser.js` style, by grouping them into a utility file.